src/qutip_qip/circuit.py

```diff
@@ -2066,6 +2072,24 @@ def png(self):
         """
         return DisplayImage(self._raw_png(), embed=True)
 
+    def draw(
+        self,
+        file_type="png",
+        density_dpi=100,
+        file_name="exported_pic",
+        file_path="",
+    ):
+        if file_type == "svg":
+            mode = "w"
+            image_data = self._raw_svg()
+        else:
+            mode = "wb"
+            image_data = self._raw_img(file_type, density_dpi)
```

Comment:
We should be able to remove this if else branch and just use `_raw_img`.

Reply:
We don't need if else branch to get `image_data` now but, we still need if else to assign write modes for SVG & PNG.

Reply:
Ah, yes that's right.